watertap/flowsheets/activated_sludge/ASM1_flowsheet.py

```diff
@@ -344,3 +307,227 @@ def function(unit):
+        "S_NO",
+        "S_NH",
+        "S_ND",
+        "X_ND",
+    ]
+
+    # composition values for treated effluent after METAB
+    comp_vals = [
+        0.086312003,
+        0.039661279,
+        0.11152474,
+        0.749114341,
+        1.40e-10,
+        1.40e-10,
+        1.40e-10,
+        1.40e-10,
+        1.40e-10,
+        0.205778397,
+        0.005252493,
+        0.058108176,
+    ]
+
+    m.fs.R3.KLa.fix(20)
+    m.fs.R4.KLa.fix(20)
+    m.fs.R5.KLa.fix(20)
+    new_comps = {comps[i]: comp_vals[i] for i in range(len(comps))}
+    solver = get_solver()
+    # solve_stat = {}
+    # clones = {}
+    # for i, (k,v) in enumerate(new_comps.items()):
+    #     clone_name = f"clone_change_{k}"
+    #     m_clone = m.clone()
+    #     old_val = pyo.value(m_clone.fs.feed.conc_mass_comp[0,k])
+    #     m_clone.fs.feed.conc_mass_comp[0,k].fix(v)
+    #     print(f"{k} was fixed.")
+    #     res = solver.solve(m_clone, tee=True)
+    #     if pyo.check_optimal_termination(res):
+    #         solve_stat[k]=1
+    #     else:
+    #         solve_stat[k]=0
+
+    #     clones[clone_name] = m_clone
+    #     m_clone.fs.feed.conc_mass_comp[0,k].fix(old_val)
+    #     del m_clone
+
+    # assert False
+    for i, (k, v) in enumerate(new_comps.items()):
+        print(f"{k} was fixed.")
+        m.fs.feed.conc_mass_comp[0, k].fix(v)
+
+    res = solver.solve(m, tee=True)
+
+    if pyo.check_optimal_termination(res):
+        print("Success!")
+    else:
+        raise RuntimeError("Failed to solve")
+    m.fs.feed.alkalinity.fix(47.7 * pyo.units.mol / pyo.units.m**3)
+    m.fs.feed.temperature.fix(308)
+    for var in m.fs.component_data_objects(pyo.Var, descend_into=True):
+        # if "flow_vol" in var.name:
+        #     iscale.set_scaling_factor(var, 1e1)
+        # if "temperature" in var.name:
+        #     iscale.set_scaling_factor(var, 1e-1)
+        # if "pressure" in var.name:
+        #     iscale.set_scaling_factor(var, 1e-3)
+        if "conc_mass_comp" in var.name:
+            print(var.name)
+            iscale.set_scaling_factor(var, 10 / (pyo.value(var) + 1e-6))
+    iscale.calculate_scaling_factors(m.fs)
+
+    m.fs.R3.KLa.fix(20)
+    m.fs.R4.KLa.fix(20)
+    m.fs.R5.KLa.fix(20)
+    # m.fs.feed.flow_vol.fix(0.00011 * 1e4
+    #                     #    134500221723699
+    #                        )
+
+    # for var in m.fs.component_data_objects(pyo.Var, descend_into=True):
+    #     if "flow_vol" in var.name:
+    #         iscale.set_scaling_factor(var, 1* pyo.value(var))
+    # m.fs.R1.volume.fix(1000 *5e-4 * pyo.units.m**3)
+    # m.fs.R2.volume.fix(1000 *5e-4 * pyo.units.m**3)
+    # m.fs.R3.volume.fix(1333 *5e-4* pyo.units.m**3)
+    # m.fs.R4.volume.fix(1333 *5e-4* pyo.units.m**3)
+    # m.fs.R5.volume.fix(1333 *5e-4* pyo.units.m**3)
+    # for i in range(5):
+    #     blk = getattr(m.fs,f"R{i+1}")
+    #     iscale.set_scaling_factor(blk.volume, 1e2)
+    # iscale.calculate_scaling_factors(m)
```

Comment:
Tried lower influent flowrates but could not solve at 0.001 or 0.0001 (0.01 and higher did converge however). I also tried adjusting reactor volumes accordingly, with no success.